### PR TITLE
use babel 7, updated deps

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+        "targets": {
+          "ie": 11
+        }
+      }
+    ]
+  ],
+  "plugins": [
+    "@babel/plugin-proposal-object-rest-spread",
+    "transform-class-properties"
+  ]
```
Comment (lennym) [inline .babelrc]: Does this work if we rename it to `.babelrc.json` and if so, can we do that and then `require` it when we use the same config elsewhere?

Comment (joefitter) [reply .babelrc]: seems to 👍

File: lib/register.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
-          node: 'current'
+          ie: 11
         }
       }
     ]
   ],
-  plugins: 'transform-object-rest-spread'
+  plugins: [
+    '@babel/plugin-proposal-object-rest-spread'
+  ]
```
Comment (lennym) [inline lib/register.js]: Can we replace all (or most) of this config with a `require('../.babel.json')`?

File: ui/webpack.config.js [hunk 3]
```diff
@@ -59,7 +59,27 @@ module.exports = dirs => {
...
+                    }
+                  }
+                ]
+              ],
+              plugins: [
+                '@babel/plugin-proposal-object-rest-spread',
+                'transform-class-properties'
+              ]
+            }
+          }
```
Comment (lennym) [inline ui/webpack.config.js]: As above re `require`-ing the config.

**Pull request merged**